integration-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This crate contains the common code for integration tests
+
+#![deny(clippy::all)]
+#![deny(missing_docs)]
+
+use std::path::PathBuf;
+use std::process::Command;
+use std::time::Duration;
+use wait_timeout::ChildExt;
+
+#[cfg_attr(not(any(has_sgx, has_sev)), ignore)]
+pub fn keep_under_test() -> &'static str {
+    let mut result = "";
+    if cfg!(has_sgx) {
+        result = "enarx-keep-sgx";
+    }
+    if cfg!(has_sev) {
+        result = "enarx-keep-sev";
+    }
+    result
+}
+
+pub struct IntegrationTest {
+    pub wksp_root: PathBuf,
+    pub keep: PathBuf,
+    pub payload: PathBuf,
+    pub shim: PathBuf,
+}
+
+impl IntegrationTest {
+    pub fn new(bin: &str) -> Self {
+        let enarx_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let enarx_dir = enarx_dir.parent().unwrap();
+        let keep_type = keep_under_test();
+        Self {
+            wksp_root: enarx_dir.to_path_buf(),
+            keep: find_filepath(
+                enarx_dir
+                    .join(format!("{}/target/", keep_type))
+                    .to_path_buf(),
+                format!("{}", keep_type).as_str(),
+            )
+            .unwrap(),
+            payload: find_filepath(
+                enarx_dir
+                    .join("integration-tests/target/x86_64-unknown-linux-musl/")
+                    .to_path_buf(),
+                bin,
+            )
+            .unwrap(),
+            shim: find_filepath(
+                enarx_dir
+                    .join(format!(
+                        "{}-shim/target/x86_64-unknown-linux-musl/",
+                        keep_type
+                    ))
+                    .to_path_buf(),
+                format!("{}-shim", keep_type).as_str(),
+            )
+            .unwrap(),
+        }
+    }
+    pub fn run_test(
+        self: Self,
+        timeout: Option<usize>,
+        exit_status: Option<i32>,
+        _stdin: Option<&str>,
+        _stdout: Option<&str>,
+        _stderr: Option<&str>,
+    ) -> () {
```

Comment:
```suggestion
    }

    pub fn run_test(
        self: Self,
        timeout: usize,
        exit_status: i32,
    ) -> () {
```

1. Vertical space between functions.
2. Remove unused parameters. Just add them later.
3. Require all arguments.

In particular, we will *always* test for all the test criteria. So force the caller to think carefully about sensible values.